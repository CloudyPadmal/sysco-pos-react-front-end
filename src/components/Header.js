import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export function ConditionalHeader(P) {
    if (P.ISLOGGEDIN) {
        return (
            <div className="collapse navbar-collapse">
                <Link className="navbar-brand" style={{ paddingTop: '0px' }} to="/home">Sysco PoS System</Link>
                <div className="navbar-nav mr-auto mt-2 mt-lg-0">
                    <Link className="nav-item nav-link" to="/my_orders"><i className="fas fa-list-ul"></i> My Orders</Link>
                    <Link className="nav-item nav-link" to="/create_order"><i className="fas fa-plus"></i> Create Order</Link>
                </div>
                <div className="navbar-nav my-2 my-lg-0">
                    <Link className="nav-item nav-link" to="/logout"><i className="fas fa-sign-out-alt"></i> Logout</Link>
                </div>
            </div>
        );
    } else {
        return (
            <div className="collapse navbar-collapse">
                <Link className="navbar-brand" style={{ paddingTop: '0px' }} to="/home">Sysco PoS System</Link>
                <div className="navbar-nav ml-auto">
                    <Link className="nav-item nav-link" to="/login"><i className="fas fa-sign-in-alt"></i> Login</Link>
                    <Link className="nav-item nav-link" to="/register">Register</Link>
                </div>
            </div>
        );
    }
}

/**
 * @abstract Navigation bar at the top of page
 * @description This will change state depending on the user log state
 * @param ISLOGGEDIN Logged status from global app state
 */
class Header extends Component {
    render() {
        return (
            <header>
                <nav className="navbar navbar-expand-lg navbar-light bg-light">
                    <ConditionalHeader ISLOGGEDIN={this.props.ISLOGGEDIN} />
                </nav>
            </header>
        );
    }
}

Header.propTypes = {
    ISLOGGEDIN: PropTypes.bool,
};

export default Header;
