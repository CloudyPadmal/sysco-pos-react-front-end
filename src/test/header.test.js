import React from 'react';
import { shallow, mountWrap } from './enzyme';

import Header from '../components/Header';

describe('Navigation header', () => {

  it('renders the header as expected', () => {
    const wrapper = shallow(<Header ISLOGGEDIN={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders login header if user is not logged in', () => {
    const wrapper = mountWrap(<Header ISLOGGEDIN={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders login header if user is logged in', () => {
    const wrapper = mountWrap(<Header ISLOGGEDIN={true} />);
    expect(wrapper).toMatchSnapshot();
  });

});