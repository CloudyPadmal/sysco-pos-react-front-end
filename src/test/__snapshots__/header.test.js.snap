// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation header renders login header if user is logged in 1`] = `
<Header
  ISLOGGEDIN={true}
>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <ConditionalHeader
        ISLOGGEDIN={true}
      >
        <div
          className="collapse navbar-collapse"
        >
          <Link
            className="navbar-brand"
            replace={false}
            style={
              Object {
                "paddingTop": "0px",
              }
            }
            to="/home"
          >
            <a
              className="navbar-brand"
              href="/home"
              onClick={[Function]}
              style={
                Object {
                  "paddingTop": "0px",
                }
              }
            >
              Sysco PoS System
            </a>
          </Link>
          <div
            className="navbar-nav mr-auto mt-2 mt-lg-0"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/my_orders"
            >
              <a
                className="nav-item nav-link"
                href="/my_orders"
                onClick={[Function]}
              >
                <i
                  className="fas fa-list-ul"
                />
                 My Orders
              </a>
            </Link>
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/create_order"
            >
              <a
                className="nav-item nav-link"
                href="/create_order"
                onClick={[Function]}
              >
                <i
                  className="fas fa-plus"
                />
                 Create Order
              </a>
            </Link>
          </div>
          <div
            className="navbar-nav my-2 my-lg-0"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/logout"
            >
              <a
                className="nav-item nav-link"
                href="/logout"
                onClick={[Function]}
              >
                <i
                  className="fas fa-sign-out-alt"
                />
                 Logout
              </a>
            </Link>
          </div>
        </div>
      </ConditionalHeader>
    </nav>
  </header>
</Header>
`;

exports[`Navigation header renders login header if user is not logged in 1`] = `
<Header
  ISLOGGEDIN={false}
>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <ConditionalHeader
        ISLOGGEDIN={false}
      >
        <div
          className="collapse navbar-collapse"
        >
          <Link
            className="navbar-brand"
            replace={false}
            style={
              Object {
                "paddingTop": "0px",
              }
            }
            to="/home"
          >
            <a
              className="navbar-brand"
              href="/home"
              onClick={[Function]}
              style={
                Object {
                  "paddingTop": "0px",
                }
              }
            >
              Sysco PoS System
            </a>
          </Link>
          <div
            className="navbar-nav ml-auto"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/login"
            >
              <a
                className="nav-item nav-link"
                href="/login"
                onClick={[Function]}
              >
                <i
                  className="fas fa-sign-in-alt"
                />
                 Login
              </a>
            </Link>
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/register"
            >
              <a
                className="nav-item nav-link"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </Link>
          </div>
        </div>
      </ConditionalHeader>
    </nav>
  </header>
</Header>
`;

exports[`Navigation header renders the header as expected 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <ConditionalHeader
      ISLOGGEDIN={true}
    />
  </nav>
</header>
`;
